src/Interpreters/ProcessList.cpp

```diff
@@ -428,25 +436,72 @@ QueryStatus * ProcessList::tryGetProcessListElement(const String & current_query
 }
 
 
-CancellationCode ProcessList::sendCancelToQuery(const String & current_query_id, const String & current_user, bool kill)
+CancellationCode ProcessList::sendCancelToQuery(const String & current_query_id, const String & current_user)
 {
-    auto lock = safeLock();
+    auto [lock, blocker] = safeLock();
 
     QueryStatus * elem = tryGetProcessListElement(current_query_id, current_user);
+    // To forbid element to be removed from process list
+    elem->forbidRemoval();
```

Comment:
> To forbid element to be removed from process list

I see it from the method name, the comment should explain **why** do we forbid it